### PR TITLE
handling randomizing of retries and erc20 failures

### PR DESCRIPTION
**Pull request by julien51**
# Description

Thanks Peter from MetaCartel for helping find that!

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [X] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [X] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-js/src/FetchJsonProvider.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ export default class FetchJsonProvider extends providers.JsonRpcProvider {
           if (error.statusCode === 429 && tries < this.maxRetries) {
             return setTimeout(() => {
               sendOnce(tries + 1)
-            }, this.rateLimit)
+            }, Math.floor(this.rateLimit + (Math.random() * this.rateLimit) / 4))
```
Comment (julien51) [inline unlock-js/src/FetchJsonProvider.js]: This is not strictly required here but will make our retries work better.

File: unlock-js/src/erc20.js [hunk 2]
```diff
@@ -32,8 +38,14 @@ export async function getErc20Decimals(erc20ContractAddress, provider) {
...
 export async function getErc20TokenSymbol(erc20ContractAddress, provider) {
   const contract = new ethers.Contract(erc20ContractAddress, erc20abi, provider)
-  const symbolPromise = await contract.symbol()
-  return symbolPromise
+  let symbol
+  try {
+    symbol = await contract.symbol()
+  } catch (e) {
+    /** Some ERC20 contracts, including DAI do not have the right symbol method. */
+    return null
```
Comment (cnasc) [inline unlock-js/src/erc20.js]: What will we show on the paywall in this case?

Comment (akeem) [reply unlock-js/src/erc20.js]: Should we do a hardcode for the DAI address as a stopgap?

Comment (julien51) [reply unlock-js/src/erc20.js]: On the paywall, DAI is hardcoded for its adress, and others will show ERC20.
We can probably move that logic here at some point.

**Pull request merged**